mutacc/builds/build_variant.py

```diff
@@ -18,19 +18,40 @@ class Variant(dict):
         Class to represent variant
     """
 
-    def __init__(self, vcf_entry, samples, padding, parser=None):
+    def __init__(self, vcf_entry, samples, padding, sv_padding, parser=None):
 
         super(Variant, self).__init__()
 
         self.entry = vcf_entry
         self.samples = samples
-
-        self.build_variant_object(padding)
-        self.entry = str(self.entry)
+        self.padding = padding
+        self.sv_padding = sv_padding
+        self.build_variant_object()
 
         if parser is not None:
             self.update(parser.parse(vcf_entry))
 
+        del self.entry
+
+    def _get_padding(self, variant_type):
+
+        if variant_type == "SNV":
+            return self.padding
+
+        if self.entry.INFO.get("CIPOS") or self.entry.INFO.get("CIEND"):
+
+            cis = []
+            if self.entry.INFO.get("CIPOS"):
+                cis.append(abs(self.entry.INFO["CIPOS"][0]))
+            if self.entry.INFO.get("CIEND"):
+                cis.append(abs(self.entry.INFO["CIEND"][-1]))
+            ci_max = max(cis)
```

Comment:
Is this confidence interval?

Reply:
yes, CIPOS and CIEND are given by some SV callers, giving a confidence interval e.g. CIPOS=-100,100;CIEND=-50,50 around the start and beginning. I wanted to incorporate this into the padding for the SVs that have these flags present.